private.tf

```diff
@@ -0,0 +1,55 @@
+module "private_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=tags/0.2.2"
+  namespace  = "${var.namespace}"
+  name       = "${var.availability_zone}"
+  attributes = ["private"]
+  stage      = "${var.stage}"
+  delimiter  = "${var.delimiter}"
+  tags       = "${var.tags}"
+}
+
+resource "aws_subnet" "private" {
+  count             = "${var.type == "private" ? length(var.names) : 0}"
+  vpc_id            = "${var.vpc_id}"
+  availability_zone = "${var.availability_zone}"
+  cidr_block        = "${cidrsubnet(var.cidr_block, ceil(log(length(var.names), 2)), count.index)}"
+
+  tags = {
+    "Name"      = "${module.private_label.id}${var.delimiter}${element(var.names, count.index)}"
+    "Stage"     = "${module.private_label.stage}"
+    "Namespace" = "${module.private_label.namespace}"
+  }
```

Comment:
necessary to have some tag for AZ for case when using same names of subnets in different AZ

Reply:
@SweetOps please have a look onto https://github.com/cloudposse/terraform-aws-named-subnets/pull/3#issuecomment-339398771